### PR TITLE
[core] Use APIError in more places

### PR DESCRIPTION
**Pull request by maxsharabayko**
A set of changes extracted from #1244, that does not affect the functionality. Code improvement.

The second commit contains minor fixes of some issues reported by [SonarCloud](https://sonarcloud.io/project/issues?createdAfter=2020-03-01&createdBefore=2020-04-23&directories=srtcore&id=srt&open=AXDKvvUnZJZS4c4MaLuj&resolved=false) static analyzer.

### CODE REVIEW

File: srtcore/api.cpp [hunk 1]
```diff
@@ -2980,7 +2973,7 @@ SRTSOCKET CUDT::accept(SRTSOCKET u, sockaddr* addr, int* addrlen)
       s_UDTUnited.setError(new CUDTException(e));
       return INVALID_SOCK;
    }
-   catch (bad_alloc&)
+   catch (const bad_alloc&)
```
Comment (ethouris) [inline srtcore/api.cpp:2976]: Adding const here is superfluous, provided there's no name declared here. It's kinda neutral and in this case changes nothing.

Comment (maxsharabayko) [reply srtcore/api.cpp:2976]: In general, I agree that this is superfluous. This is reported by SonarCloud: [link](https://sonarcloud.io/project/issues?id=srt&issues=AXDFjdaG1BT7hYfjdXR3&open=AXDFjdaG1BT7hYfjdXR3). Although there is no variable declared, still I think nothing bad from extra const, that ensures proper casting.

File: srtcore/api.cpp [hunk 2]
```diff
@@ -183,7 +183,6 @@ m_GCStopCond(),
 m_InitLock(),
 m_iInstanceCount(0),
 m_bGCStatus(false),
-m_GCThread(),
```
Comment (maxsharabayko) [inline srtcore/api.cpp:186]: - [ ] Revert this.

File: srtcore/core.cpp [hunk 3]
```diff
@@ -465,14 +465,14 @@ void CUDT::setOpt(SRT_SOCKOPT optName, const void* optval, int optlen)
...
-        if (*(int *)optval <= 0)
+        if (*(const int *)optval <= 0)
             throw CUDTException(MJ_NOTSUP, MN_INVAL, 0);
 
         {
             // This weird cast through int is required because
             // API requires 'int', and internals require 'size_t';
             // their size is different on 64-bit systems.
-            size_t val = size_t(*(int *)optval);
+            const size_t val = size_t(*(const int *)optval);
```
Comment (maxsharabayko) [inline srtcore/core.cpp:475]: @ethouris 
>There is one case inn commit 2 that makes sense - it applies const to a size_t variable of something

This one?

Comment (ethouris) [reply srtcore/core.cpp:475]: Yes.

**Pull request merged**